[16.0][ADD] pos_membership_delegated_partner

flaenen opened the pull request:
WIP

This module implements the features of the Membership Delegated Partner module in the point of sale UI. This allows a customer to purchase memberships for other partners.

### Dependencies

- pos_membership
- https://github.com/OCA/vertical-association/pull/207

### Usage

1. Open the point of sale
2. Add a membership product to the order
3. Select the corresponding order line
4. Click on the "Delegated Member" button that has just appeared
5. Select a partner from the screen that opens

Once the order has been validated, the designated partner becomes a member.

huguesdk commented on pos_membership_delegated_partner/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+        "point_of_sale.assets": [
+            "pos_membership_delegated_partner/static/src/css/*.css",
+            "pos_membership_delegated_partner/static/src/js/*.js",
+            "pos_membership_delegated_partner/static/src/xml/*.xml",
```
i think it’s best to have js and xml files in the corresponding directory of the files they extend. for example, `ProductScreen.esm.js` should go to `static/src/js/Screen/ProductScreen/ProductScreen.esm.js` and `OrderlineDetails.xml` should go to `static/src/xml/Screens/TicketScreen/OrderlineDetails.xml`.

huguesdk commented on pos_membership_delegated_partner/static/src/xml/Orderline.xml:
```diff
@@ -0,0 +1,42 @@
+    >
+        <xpath expr="//ul[hasclass('info-list')]" position="inside">
+            <li class="info orderline-membership" t-if="props.line.product.membership">
+                <t t-if="props.line.delegated_member">
+                    <i
+                        class="fa fa-hand-o-right"
+                        role="img"
+                        aria-label="Member"
+                        title="Member"
+                    />
+                    <t t-out="props.line.delegated_member.name" />
+                </t>
+                <t t-else="">
+                    <t t-if="props.line.order.get_partner()">
+                        <i
+                            class="fa fa-hand-o-right"
+                            role="img"
+                            aria-label="Member"
+                            title="Member"
+                        />
+                        <t t-out="props.line.order.get_partner().name" />
+                        <span> (<i
+                                class="fa fa-user"
+                                role="img"
+                                aria-label="Customer"
+                                title="Customer"
+                            />)</span>
+                    </t>
+                </t>
```
instead of having a big `t-if`/`t-else` with almost the same thing in both parts, the orderline should have a method returning the membership partner (with the logic there). the display of the customer icon could be in an `t-if="!props.line.delegated_member"`.

huguesdk commented on pos_membership_delegated_partner/static/src/xml/OrderlineDetails.xml:
```diff
@@ -0,0 +1,42 @@
+                                aria-label="Customer"
+                                title="Customer"
+                            />)</span>
+                    </t>
```
same remark as in the `Orderline` template. better: as these have exactly the same content, this should additionally be defined in a component reused in both places.

huguesdk reviewed (changes requested):
some remarks, mostly cosmetic. otherwise, lgtm.

flaenen committed:
[ADD] pos_membership_delegated_partner
This module implements the features of the Membership Delegated Partner module
in the point of sale UI. This allows a customer to purchase memberships for
other partners.

flaenen commented:
@huguesdk Thanks for the review! I've made the adjustments.

huguesdk reviewed (approved):
thanks for the changes! lgtm! 🎉
